tests/components/pushbullet/test_notify.py

```diff
@@ -1,94 +1,38 @@
-"""The tests for the pushbullet notification platform."""
+"""Test pushbullet notification platform."""
 from http import HTTPStatus
-import json
-from unittest.mock import patch
 
 from pushbullet import PushBullet
 import pytest
+from requests_mock import Mocker
 
-import homeassistant.components.notify as notify
-from homeassistant.setup import async_setup_component
+from homeassistant.components.pushbullet.notify import PushBulletNotificationService
 
-from tests.common import assert_setup_component, load_fixture
 
+@pytest.fixture()
+def mock_service(hass):
+    """Return Mock PushBulletNotificationService."""
+    yield PushBulletNotificationService(hass, PushBullet("MYAPIKEY"))
 
-@pytest.fixture
-def mock_pushbullet():
-    """Mock pushbullet."""
-    with patch.object(
-        PushBullet,
-        "_get_data",
-        return_value=json.loads(load_fixture("devices.json", "pushbullet")),
-    ):
-        yield
 
-
-async def test_pushbullet_config(hass, mock_pushbullet):
-    """Test setup."""
-    config = {
-        notify.DOMAIN: {
-            "name": "test",
-            "platform": "pushbullet",
-            "api_key": "MYFAKEKEY",
-        }
-    }
-    with assert_setup_component(1) as handle_config:
-        assert await async_setup_component(hass, notify.DOMAIN, config)
-        await hass.async_block_till_done()
-    assert handle_config[notify.DOMAIN]
-
-
-async def test_pushbullet_config_bad(hass):
-    """Test set up the platform with bad/missing configuration."""
-    config = {notify.DOMAIN: {"platform": "pushbullet"}}
-    with assert_setup_component(0) as handle_config:
-        assert await async_setup_component(hass, notify.DOMAIN, config)
-        await hass.async_block_till_done()
-    assert not handle_config[notify.DOMAIN]
-
-
-async def test_pushbullet_push_default(hass, requests_mock, mock_pushbullet):
+async def test_pushbullet_push_default(hass, requests_mock: Mocker, mock_service):
     """Test pushbullet push to default target."""
-    config = {
-        notify.DOMAIN: {
-            "name": "test",
-            "platform": "pushbullet",
-            "api_key": "MYFAKEKEY",
-        }
-    }
-    with assert_setup_component(1) as handle_config:
-        assert await async_setup_component(hass, notify.DOMAIN, config)
-        await hass.async_block_till_done()
-    assert handle_config[notify.DOMAIN]
     requests_mock.register_uri(
         "POST",
         "https://api.pushbullet.com/v2/pushes",
         status_code=HTTPStatus.OK,
         json={"mock_response": "Ok"},
     )
     data = {"title": "Test Title", "message": "Test Message"}
-    await hass.services.async_call(notify.DOMAIN, "test", data)
+    mock_service.send_message(**data)
```

Comment:
We want to call the service, not access the notify instance directly.

Reply:
I agree, but when I try to set up the entry and call the service I get an error in the tests related to frozen threads. As you can see this integration uses threading so I would appreciate if you can take a look and let me if the implementation of threading is valid here and in this case how should I handle it in the tests.